From 3707f467f9a26a7df3d41385023b43c3d08911d2 Mon Sep 17 00:00:00 2001
From: Manjukumar Matha <manjukumar.harthikote-matha@xilinx.com>
Date: Tue, 12 Feb 2019 17:46:52 -0800
Subject: [PATCH][v3] CMakeLists.txt: Use SWIG_SUPPORT_FILES to find the list
 of generated files for cmake version 3.12 or higher

Use SWIG_SUPPORT_FILES to find the list of python files generated by
CMake Swig module and install those files. This should be applicable to
cmake version 3.12 or higher

Signed-off-by: Manjukumar Matha <manjukumar.harthikote-matha@xilinx.com>
---
 src/CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e19cda2..b565814 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -337,8 +337,14 @@ macro(_upm_swig_python)
       OUTPUT_NAME _pyupm_${libname}
       LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_PYTHON_BINARY_DIR})
 
+  if (CMAKE_VERSION VERSION_LESS "3.12")
+     set(support_files ${swig_extra_generated_files})
+  else()
+     get_property(support_files TARGET _${python_wrapper_name} PROPERTY SWIG_SUPPORT_FILES)
+  endif()
+
   # Install .py's to python packages directory/upm
-  install (FILES ${swig_extra_generated_files}
+  install (FILES ${support_files}
       DESTINATION ${PYTHON_PACKAGES_PATH}/upm
       COMPONENT ${CMAKE_PROJECT_NAME}-python${PYTHON_VERSION_MAJOR})
 
-- 
2.7.4

